homeassistant/components/plex/server.py

```diff
@@ -78,13 +88,92 @@ def _connect_with_url():
+                "Could not connect to Plex server: %s (%s)", self.friendly_name, ex
+            )
+            return
+
+        for device in devices:
+            available_client_ids.append(device.machineIdentifier)
+
+            if device.machineIdentifier not in self._clients:
+                new_client = PlexMediaPlayer(
+                    self, device, None, self._sessions, self.update_platforms
+                )
+                self._clients[device.machineIdentifier] = new_client
+                _LOGGER.debug("New device: %s", device.machineIdentifier)
+                new_plex_clients.append(new_client)
+            else:
+                _LOGGER.debug("Refreshing device: %s", device.machineIdentifier)
+                self._clients[device.machineIdentifier].refresh(device, None)
+
+        try:
+            sessions = self._plex_server.sessions()
+        except plexapi.exceptions.BadRequest:
+            _LOGGER.exception("Error listing Plex sessions")
+            return
+        except requests.exceptions.RequestException as ex:
+            _LOGGER.warning(
+                "Could not connect to Plex server: %s (%s)", self.friendly_name, ex
+            )
+            return
+
+        self._sessions.clear()
+        for session in sessions:
+            for player in session.players:
+                self._sessions[player.machineIdentifier] = session, player
+
+        for machine_identifier, (session, player) in self._sessions.items():
+            if machine_identifier in available_client_ids:
+                # Avoid using session if already added as a device.
+                _LOGGER.debug("Skipping session, device exists: %s", machine_identifier)
+                continue
+
+            if (
+                machine_identifier not in self._clients
+                and machine_identifier is not None
+            ):
+                new_client = PlexMediaPlayer(
+                    self, player, session, self._sessions, self.update_platforms
+                )
+                self._clients[machine_identifier] = new_client
+                _LOGGER.debug("New session: %s", machine_identifier)
+                new_plex_clients.append(new_client)
+            else:
+                _LOGGER.debug("Refreshing session: %s", machine_identifier)
+                self._clients[machine_identifier].refresh(None, session)
+
+        for client in self._clients.values():
+            # force devices to idle that do not have a valid session
+            if client.session is None:
+                client.force_idle()
+
+            client.set_availability(
+                client.machine_identifier in available_client_ids
+                or client.machine_identifier in self._sessions
+            )
+
+            if client not in new_plex_clients:
+                client.schedule_update_ha_state()
+
+        if new_plex_clients:
+            self.add_media_player_callback(  # pylint: disable=not-callable
```

Comment:
Signal the platform to add an entity via dispatch helper. See eg the zha integration.